### PR TITLE
fix(frontend): forgot-password reset-password forms validation bb-285

### PR DESCRIPTION
**Pull request by ihorLysak**
### Summary

Rewritten validation to rely on validation schema, moved a constant which is used in auth forms only closer to these forms.

### CODE REVIEW

File: apps/frontend/src/pages/quiz/quiz.tsx
Comment (ihorLysak) [inline apps/frontend/src/pages/quiz/quiz.tsx:1]: Moved these around so they are in the same order as their steps

File: apps/frontend/src/pages/auth/libs/components/forgot-password-form/forgot-password-form.tsx [hunk 1]
```diff
@@ -17,6 +18,7 @@ const ForgotPasswordForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 		defaultValues: DEFAULT_FORGOT_PASSWORD_PAYLOAD,
 		validationSchema: userForgotPasswordValidationSchema,
 	});
+	const isInputError = Object.keys(errors).length > NO_ERROR_INPUT_FIELD_AMOUNT;
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/forgot-password-form/forgot-password-form.tsx:21]:
```suggestion
	const hasError = Object.keys(errors).length > NO_ERROR_INPUT_FIELD_AMOUNT;
```

File: apps/frontend/src/pages/auth/libs/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { Button, Input, Link } from "~/libs/components/components.js";
 import { AppRoute } from "~/libs/enums/enums.js";
 import { useAppForm, useCallback, useState } from "~/libs/hooks/hooks.js";
-import { NO_ERROR_INPUT_FIELD_AMOUNT } from "~/modules/users/libs/constants/constants.js";
 import {
 	type UserSignInRequestDto,
 	userSignInValidationSchema,
 } from "~/modules/users/users.js";
 
+import { NO_ERROR_INPUT_FIELD_AMOUNT } from "../../constants/constants.js";
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/sign-in-form/sign-in-form.tsx:9]: Use `NumericValue`

File: apps/frontend/src/pages/quiz/quiz.tsx [hunk 3]
```diff
@@ -35,22 +35,22 @@ const Quiz: React.FC = () => {
...
 
-			case Step.BALANCE_WHEEL: {
-				return <BalanceWheel />;
-			}
-
 			case Step.INTRODUCTION: {
 				return <Introduction onNext={handleNextStep} />;
 			}
 
+			case Step.QUIZ: {
```
Comment (fshabanov) [inline apps/frontend/src/pages/quiz/quiz.tsx:42]: Try to avoid the changes, that are not required in your PR

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/quiz/quiz.tsx (hunk 3 above)
Comment (ihorLysak) [reply apps/frontend/src/pages/quiz/quiz.tsx:42]: Somehow these changes are already on main

**Comment by ihorLysak** [comment]
Using NumericalValue enum in validations, defined a copy of it inside of shared. When this branch gets merged, will ask mobile team to update imports so that they get this enum from shared, this will be safer than doing it myself.

**Review by fshabanov — APPROVED** [review]

**Pull request merged**